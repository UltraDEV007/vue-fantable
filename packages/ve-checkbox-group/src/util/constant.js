// prefix
export const PREFIX_CLS = 've-checkbox-group-'

// emit events
export const EMIT_EVENTS = {
  // on checked change
  ON_CHECKED_CHANGE: 'on-checked-change',
}

// comps name
export const COMPS_NAME = {
  VE_CHECKBOX: 'VeCheckbox',
  VE_CHECKBOX_GROUP: 'VeCheckboxGroup',
}
