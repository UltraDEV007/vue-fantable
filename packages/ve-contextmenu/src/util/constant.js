// prefix
export const PREFIX_CLS = 've-contextmenu-'

// comps name
export const COMPS_NAME = {
  VE_CONTEXTMENU: 'VeContextmenu',
}

// emit events
export const EMIT_EVENTS = {
  ON_NODE_CLICK: 'on-node-click',
}

// init data
export const INIT_DATA = {
  // PARENT_ID: 0,
  PARENT_DEEP: -1,
}

// contextmenu types
export const CONTEXTMENU_NODE_TYPES = {
  // Separator
  SEPARATOR: 'SEPARATOR',
}

// instance methods
export const INSTANCE_METHODS = {
  HIDE_CONTEXTMENU: 'hideContextmenu',
}
