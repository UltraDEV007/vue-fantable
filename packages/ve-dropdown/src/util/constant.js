// prefix
export const PREFIX_CLS = 've-dropdown-'

// emit events
export const EMIT_EVENTS = {
  FILTER_RESET: 'on-filter-reset',
  FILTER_CONFIRM: 'on-filter-confirm',
  VISIBLE_CHANGE: 'on-dropdown-visible-change',
  ITEM_SELECT_CHANGE: 'on-item-select-change',
}

// comps name
export const COMPS_NAME = {
  VE_DROPDOWN: 'VeDropdown',
}
