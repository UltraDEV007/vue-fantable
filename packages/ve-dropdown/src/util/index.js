/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return 've-dropdown-' + cls
}
