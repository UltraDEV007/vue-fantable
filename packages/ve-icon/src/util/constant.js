// prefix
export const PREFIX_CLS = 've-icon-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_ICON: 'VeIcon',
}
