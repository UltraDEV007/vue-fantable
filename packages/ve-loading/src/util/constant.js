// prefix
export const PREFIX_CLS = 've-loading-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_LOADING: 'VeLoading',
  VE_LOADING_PLANE: 'VeLoadingPlane',
  /* VE_LOADING_CHASE: "VeLoadingChase", */
  VE_LOADING_BOUNCE: 'VeLoadingBounce',
  VE_LOADING_WAVE: 'VeLoadingWave',
  VE_LOADING_PULSE: 'VeLoadingPulse',
  VE_LOADING_FLOW: 'VeLoadingFlow',
  /* VE_LOADING_SWING: "VeLoadingSwing", */
  /*  VE_LOADING_CIRCLE: "VeLoadingCircle",
VE_LOADING_CIRCLE_FADE: "VeLoadingCircleFade", */
  VE_LOADING_GRID: 'VeLoadingGrid',
  VE_LOADING_FOLD: 'VeLoadingFold',
}

// spin name
export const SPIN_NAMES = {
  PLANE: 'plane',
  GRID: 'grid',
  WAVE: 'wave',
  FLOW: 'flow',
  BOUNCE: 'bounce',
  PULSE: 'pulse',
}
