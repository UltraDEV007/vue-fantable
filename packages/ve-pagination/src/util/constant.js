// prefix
export const PREFIX_CLS = 've-pagination-'

// locale comp name
export const LOCALE_COMP_NAME = 'pagination'

// emit events
export const EMIT_EVENTS = {
  PAGE_NUMBER_CHANGE: 'on-page-number-change',
  PAGE_SIZE_CHANGE: 'on-page-size-change',
}

// comps name
export const COMPS_NAME = {
  VE_PAGINATION: 'VePagination',
}
