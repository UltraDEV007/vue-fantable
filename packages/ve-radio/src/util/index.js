import { PREFIX_CLS } from './constant'

/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return PREFIX_CLS + cls
}
