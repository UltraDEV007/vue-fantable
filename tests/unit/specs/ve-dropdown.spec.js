import { mount } from "@vue/test-utils";
import veDropdown from "@/ve-dropdown";
import { later } from "../util";

describe("veDropdown", () => {
    // dropdown items
    const DROPDOWN_ITEMS = [
        { value: 0, label: "Apple" },
        { value: 1, label: "Orange", selected: true },
        { value: 2, label: "Banana" },
    ];

    it("render normal", () => {
        const wrapper = mount(veDropdown, {
            propsData: {
                value: DROPDOWN_ITEMS,
                showOperation: true,
                confirmFilterText: "filter",
                resetFilterText: "reset",
            },
            slots: {
                default: `<div>click me</div>`,
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("render with radio", () => {
        const wrapper = mount(veDropdown, {
            propsData: {
                value: DROPDOWN_ITEMS,
                showRadio: true,
                showOperation: true,
                confirmFilterText: "filter",
                resetFilterText: "reset",
            },
            slots: {
                default: `<div>click me</div>`,
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("render with checkbox", () => {
        const wrapper = mount(veDropdown, {
            propsData: {
                value: DROPDOWN_ITEMS,
                isMultiple: true,
                showOperation: true,
                confirmFilterText: "filter",
                resetFilterText: "reset",
            },
            slots: {
                default: `<div>click me</div>`,
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("value props", () => {
        const wrapper = mount(veDropdown, {
            propsData: {
                value: DROPDOWN_ITEMS,
            },
            slots: {
                default: `<div>click me</div>`,
            },
        });

        expect(wrapper.findAll(".ve-dropdown-items-li").length).toBe(3);
    });

    it("showOperation prop", () => {
        const wrapper = mount(veDropdown, {
            propsData: {
                value: DROPDOWN_ITEMS,
                showOperation: true,
            },
            slots: {
                default: `<div>click me</div>`,
            },
        });

        expect(wrapper.find(".ve-dropdown-operation").exists()).toBe(true);
    });

    it("isMultiple prop", () => {
        const wrapper = mount(veDropdown, {
            propsData: {
                value: DROPDOWN_ITEMS,
                isMultiple: true,
            },
            slots: {
                default: `<div>click me</div>`,
            },
        });

        expect(wrapper.find(".ve-dropdown-items-multiple").exists()).toBe(true);
    });

    it("placeholder prop", () => {
        const wrapper = mount(veDropdown, {
            propsData: {
                value: DROPDOWN_ITEMS,
                isMultiple: true,
            },
            slots: {
                default: `<div>click me</div>`,
            },
        });

        expect(wrapper.find(".ve-dropdown-items-multiple").exists()).toBe(true);
    });

    it("textAlign prop", () => {
        const wrapper = mount(veDropdown, {
            propsData: {
                value: DROPDOWN_ITEMS,
                textAlign: "center",
            },
            slots: {
                default: `<div>click me</div>`,
            },
        });

        expect(wrapper.find(".ve-dropdown-items-li-a-center").exists()).toBe(
            true,
        );
    });

    it("operation text prop", () => {
        const wrapper = mount(veDropdown, {
            propsData: {
                value: DROPDOWN_ITEMS,
                showOperation: true,
                confirmFilterText: "ok",
                resetFilterText: "cancel",
            },
            slots: {
                default: `<div>click me</div>`,
            },
        });

        expect(wrapper.find(".ve-dropdown-operation").text()).toContain("ok");
        expect(wrapper.find(".ve-dropdown-operation").text()).toContain(
            "cancel",
        );
    });

    it("showRadio prop", () => {
        const wrapper = mount(veDropdown, {
            propsData: {
                value: DROPDOWN_ITEMS,
                showRadio: true,
            },
            slots: {
                default: `<div>click me</div>`,
            },
        });

        expect(wrapper.findAll(".ve-radio").length).toBe(3);
    });

    it("isControlled prop", () => {
        const wrapper = mount(veDropdown, {
            propsData: {
                value: DROPDOWN_ITEMS,
                isControlled: true,
                visible: true,
            },
            slots: {
                default: `<div>click me</div>`,
            },
        });

        expect(wrapper.find(".ve-dropdown-dd-show").exists()).toBe(true);
    });

    it("hideByItemClick prop", async () => {
        const wrapper = mount(veDropdown, {
            propsData: {
                value: DROPDOWN_ITEMS,
                hideByItemClick: false,
            },
            slots: {
                default: `<div class="btn">click me</div>`,
            },
        });

        wrapper.find(".btn").trigger("click");
        await later(100);
        expect(wrapper.find(".ve-dropdown-dd-show").exists()).toBe(true);

        wrapper.find(".ve-dropdown-items-li").trigger("click");
        await later(150);
        expect(wrapper.find(".ve-dropdown-dd-show").exists()).toBe(true);

        wrapper.setProps({ hideByItemClick: true });
        await later(100);
        wrapper.find(".ve-dropdown-items-li").trigger("click");
        await later(150);
        expect(wrapper.find(".ve-dropdown-dd-show").exists()).toBe(false);
    });

    it("custom-content slot", () => {
        const wrapper = mount(veDropdown, {
            propsData: {
                value: DROPDOWN_ITEMS,
                isCustomContent: true,
            },
            slots: {
                "custom-content": `<div class="custom-content-slot">this is custom content</div>`,
            },
        });

        expect(wrapper.find(".custom-content-slot").exists()).toBe(true);
    });

    it("on-item-select-change emit", async () => {
        const wrapper = mount(veDropdown, {
            propsData: {
                value: DROPDOWN_ITEMS,
            },
            slots: {
                default: `<div class="btn">click me</div>`,
            },
        });

        wrapper.find(".btn").trigger("click");
        await later(100);

        // 改变选项
        wrapper.findAll(".ve-dropdown-items-li").at(1).trigger("click");

        await later();

        let items = [
            { value: 0, label: "Apple", selected: false },
            { value: 1, label: "Orange", selected: true },
            { value: 2, label: "Banana", selected: false },
        ];

        expect(wrapper.emitted("on-item-select-change")[0]).toEqual([items]);
    });

    it("visible change emit", async () => {
        const wrapper = mount(veDropdown, {
            propsData: {
                value: DROPDOWN_ITEMS,
            },
            slots: {
                default: `<div class="btn">click me</div>`,
            },
        });

        wrapper.find(".btn").trigger("click");
        await later(100);

        wrapper.find(".btn").trigger("click");
        await later(100);

        expect(wrapper.emitted("on-dropdown-visible-change").length).toEqual(2);
        expect(wrapper.emitted("on-dropdown-visible-change")[0]).toEqual([
            true,
        ]);
        expect(wrapper.emitted("on-dropdown-visible-change")[1]).toEqual([
            false,
        ]);
    });

    it("single select filter confirm emit", async () => {
        let items = [
            { value: 0, label: "Apple", selected: false },
            { value: 1, label: "Orange", selected: true },
            { value: 2, label: "Banana", selected: false },
        ];

        const wrapper = mount(veDropdown, {
            propsData: {
                value: items,
                showOperation: true,
                confirmFilterText: "ok",
                resetFilterText: "cancel",
            },
            slots: {
                default: `<div class="btn">click me</div>`,
            },
        });

        await later();

        wrapper.find(".btn").trigger("click");
        await later();

        // 改变选项
        wrapper.findAll(".ve-dropdown-items-li").at(0).trigger("click");

        await later();

        // confirm btn click
        wrapper.findAll(".ve-dropdown-operation-item").at(1).trigger("click");

        await later();

        expect(wrapper.emitted("on-filter-confirm").length).toEqual(1);
        expect(
            wrapper.emitted("on-filter-confirm")[0][0].find((x) => x.selected)
                .label,
        ).toEqual("Apple");
    });

    it("single select filter reset emit", async () => {
        let items = [
            { value: 0, label: "Apple", selected: false },
            { value: 1, label: "Orange", selected: true },
            { value: 2, label: "Banana", selected: false },
        ];

        const wrapper = mount(veDropdown, {
            propsData: {
                value: items,
                showOperation: true,
                confirmFilterText: "ok",
                resetFilterText: "cancel",
            },
            slots: {
                default: `<div class="btn">click me</div>`,
            },
        });

        await later();

        wrapper.find(".btn").trigger("click");
        await later();

        // reset btn click
        wrapper.findAll(".ve-dropdown-operation-item").at(0).trigger("click");

        await later();

        expect(wrapper.emitted("on-filter-reset").length).toEqual(1);
        expect(
            wrapper.emitted("on-filter-reset")[0][0].filter((x) => x.selected)
                .length,
        ).toEqual(0);
    });

    it("multiple select filter confirm emit", async () => {
        let items = [
            { value: 0, label: "Apple", selected: true },
            { value: 1, label: "Orange", selected: true },
            { value: 2, label: "Banana", selected: false },
        ];

        const wrapper = mount(veDropdown, {
            propsData: {
                value: items,
                isMultiple: true,
                showOperation: true,
                confirmFilterText: "ok",
                resetFilterText: "cancel",
            },
            slots: {
                default: `<div class="btn">click me</div>`,
            },
        });

        await later();

        wrapper.find(".btn").trigger("click");
        await later();

        // 改变选项
        wrapper.findAll(".ve-checkbox").at(2).trigger("click");

        await later();

        // confirm btn click
        wrapper.findAll(".ve-dropdown-operation-item").at(1).trigger("click");

        await later();

        expect(wrapper.emitted("on-filter-confirm").length).toEqual(1);
        expect(
            wrapper.emitted("on-filter-confirm")[0][0].filter((x) => x.selected)
                .length,
        ).toEqual(3);
    });

    it("multiple select filter reset emit", async () => {
        let items = [
            { value: 0, label: "Apple", selected: true },
            { value: 1, label: "Orange", selected: true },
            { value: 2, label: "Banana", selected: false },
        ];

        const wrapper = mount(veDropdown, {
            propsData: {
                value: items,
                isMultiple: true,
                showOperation: true,
                confirmFilterText: "ok",
                resetFilterText: "cancel",
            },
            slots: {
                default: `<div class="btn">click me</div>`,
            },
        });

        await later();

        wrapper.find(".btn").trigger("click");
        await later();

        // reset btn click
        wrapper.findAll(".ve-dropdown-operation-item").at(0).trigger("click");

        await later();

        expect(wrapper.emitted("on-filter-reset").length).toEqual(1);
        expect(
            wrapper.emitted("on-filter-reset")[0][0].filter((x) => x.selected)
                .length,
        ).toEqual(0);
    });

    it("beforeVisibleChange callback method", async () => {
        const mockBeforeVisibleChangeFn = jest.fn();
        const wrapper = mount({
            render() {
                return (
                    <ve-dropdown
                        hide-by-item-click={true}
                        before-visible-change={(args) =>
                            mockBeforeVisibleChangeFn(args)
                        }
                        value={DROPDOWN_ITEMS}
                        // slots= {
                        //             default: `<div class="btn">click me</div>`,
                        //         }
                    >
                        <div class="btn" style="color:blue;cursor: pointer;">
                            点击这里
                        </div>
                    </ve-dropdown>
                );
            },
        });

        wrapper.find(".btn").trigger("click");
        await later(100);
        expect(wrapper.find(".ve-dropdown-dd-show").exists()).toBe(true);

        expect(mockBeforeVisibleChangeFn).toHaveBeenCalledWith({
            nextVisible: true,
        });

        wrapper.find(".ve-dropdown-items-li").trigger("click");
        await later(150);

        expect(wrapper.find(".ve-dropdown-dd-show").exists()).toBe(false);
        expect(mockBeforeVisibleChangeFn).toHaveBeenCalledWith({
            nextVisible: false,
        });
    });
});
