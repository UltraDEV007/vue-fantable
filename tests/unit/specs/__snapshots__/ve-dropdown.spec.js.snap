// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`veDropdown render normal 1`] = `
<dl class="ve-dropdown">
  <dt class="ve-dropdown-dt"><a class="" style="width: 90px;">
      <div>click me</div>
    </a></dt>
  <div style="display: none;">
    <dd class="ve-dropdown-dd">
      <ul class="ve-dropdown-items" style="min-width: 90px; max-width: Infinitypx;">
        <div>
          <div class="ve-dropdown-items-warpper" style="max-height: 1000px;">
            <li class="ve-dropdown-items-li"><a href="javascript:void(0);" class="ve-dropdown-items-li-a ve-dropdown-items-li-a-left">Apple</a></li>
            <li class="ve-dropdown-items-li active"><a href="javascript:void(0);" class="ve-dropdown-items-li-a ve-dropdown-items-li-a-left">Orange</a></li>
            <li class="ve-dropdown-items-li"><a href="javascript:void(0);" class="ve-dropdown-items-li-a ve-dropdown-items-li-a-left">Banana</a></li>
          </div>
          <li class="ve-dropdown-operation"><a href="javascript:void(0)" class="ve-dropdown-operation-item">reset</a><a href="javascript:void(0)" class="ve-dropdown-operation-item">filter</a></li>
        </div>
      </ul>
    </dd>
  </div>
</dl>
`;

exports[`veDropdown render with checkbox 1`] = `
<dl class="ve-dropdown">
  <dt class="ve-dropdown-dt"><a class="" style="width: 90px;">
      <div>click me</div>
    </a></dt>
  <div style="display: none;">
    <dd class="ve-dropdown-dd">
      <ul class="ve-dropdown-items" style="min-width: 90px; max-width: Infinitypx;">
        <div>
          <div class="ve-dropdown-items-warpper" style="max-height: 1000px;">
            <li class="ve-dropdown-items-multiple ve-dropdown-items-li ve-dropdown-items-li-a-left"><label class="ve-checkbox" style="display: inline-block;"><span class="ve-checkbox-content"><input type="checkbox" class="ve-checkbox-input" value="Apple"><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label">Apple</span></label></li>
            <li class="ve-dropdown-items-multiple ve-dropdown-items-li ve-dropdown-items-li-a-left"><label class="ve-checkbox" style="display: inline-block;"><span class="ve-checkbox-content ve-checkbox-checked"><input type="checkbox" class="ve-checkbox-input" value="Orange"><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label">Orange</span></label></li>
            <li class="ve-dropdown-items-multiple ve-dropdown-items-li ve-dropdown-items-li-a-left"><label class="ve-checkbox" style="display: inline-block;"><span class="ve-checkbox-content"><input type="checkbox" class="ve-checkbox-input" value="Banana"><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label">Banana</span></label></li>
          </div>
          <li class="ve-dropdown-operation"><a href="javascript:void(0)" class="ve-dropdown-operation-item">reset</a><a href="javascript:void(0)" class="ve-dropdown-operation-item">filter</a></li>
        </div>
      </ul>
    </dd>
  </div>
</dl>
`;

exports[`veDropdown render with radio 1`] = `
<dl class="ve-dropdown">
  <dt class="ve-dropdown-dt"><a class="" style="width: 90px;">
      <div>click me</div>
    </a></dt>
  <div style="display: none;">
    <dd class="ve-dropdown-dd">
      <ul class="ve-dropdown-items" style="min-width: 90px; max-width: Infinitypx;">
        <div>
          <div class="ve-dropdown-items-warpper" style="max-height: 1000px;">
            <li class="ve-dropdown-items-li"><a href="javascript:void(0);" class="ve-dropdown-items-li-a ve-dropdown-items-li-a-left"><label class="ve-radio"><span class="ve-radio-container"><input type="radio" class="ve-radio-input" value=""><span class="ve-radio-inner"></span></span><span class="ve-radio-label">Apple</span></label></a></li>
            <li class="ve-dropdown-items-li active"><a href="javascript:void(0);" class="ve-dropdown-items-li-a ve-dropdown-items-li-a-left"><label class="ve-radio"><span class="ve-radio-container ve-radio-checked"><input type="radio" class="ve-radio-input" value=""><span class="ve-radio-inner"></span></span><span class="ve-radio-label">Orange</span></label></a></li>
            <li class="ve-dropdown-items-li"><a href="javascript:void(0);" class="ve-dropdown-items-li-a ve-dropdown-items-li-a-left"><label class="ve-radio"><span class="ve-radio-container"><input type="radio" class="ve-radio-input" value=""><span class="ve-radio-inner"></span></span><span class="ve-radio-label">Banana</span></label></a></li>
          </div>
          <li class="ve-dropdown-operation"><a href="javascript:void(0)" class="ve-dropdown-operation-item">reset</a><a href="javascript:void(0)" class="ve-dropdown-operation-item">filter</a></li>
        </div>
      </ul>
    </dd>
  </div>
</dl>
`;
